### PR TITLE
Add examples to Layout/IndentHash

### PR DESCRIPTION
**Pull request by seport**
Adds examples to layout/IndentHash

-----------------

Before submitting the PR make sure the following are checked:

* [x] Wrote [good commit messages][1].
* [x] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Used the same coding conventions as the rest of the project.
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [x] Added an entry to the [Changelog](../blob/master/CHANGELOG.md) if the new code introduces user-observable changes. See [changelog entry format](../blob/master/CONTRIBUTING.md#changelog-entry-format).
* [x] All tests(`rake spec`) are passing.
* [x] The new code doesn't generate RuboCop offenses that are checked by `rake internal_investigation`.
* [x] The PR relates to *only* one subject with a clear title
  and description in grammatically correct, complete sentences.
* [x] Updated cop documentation with `rake generate_cops_documentation` (required only when you've added a new cop or changed the configuration/documentation of an existing cop).

[1]: http://chris.beams.io/posts/git-commit/

### CODE REVIEW

File: lib/rubocop/cop/layout/indent_hash.rb [hunk 1]
```diff
@@ -18,8 +18,29 @@ module Layout
       #
       # This default style is called 'special_inside_parentheses'. Alternative
       # styles are 'consistent' and 'align_braces'. Here are examples:
+      # @example
+      #   # bad
+      #     special_inside_parentheses
```
Comment (mikegee) [inline lib/rubocop/cop/layout/indent_hash.rb]: Is `special_inside_parentheses` supposed to be a comment?

**Comment by bbatsov** [comment]
Can you please restyle the examples to look like https://github.com/bbatsov/rubocop/commit/a50f67d8cae615484b5033a2096b63d4d9002bd3 ?

File: lib/rubocop/cop/layout/indent_hash.rb [hunk 2]
```diff
@@ -18,8 +18,29 @@ module Layout
...
+      #     but_in_a_method_call({
+      #                            its_like: :this
+      #                           })
+      #     # align_braces
+      #     and_now_for_something = {
+      #                               completely: :different
+      #     }
       #
-      #     # special_inside_parentheses
+      #   # good
```
Comment (koic) [inline lib/rubocop/cop/layout/indent_hash.rb]: It is good to such `# bad` and `# good` comment will be aligned with indented example code. The following address is the template code.
https://github.com/bbatsov/rubocop/blob/v0.51.0/lib/rubocop/cop/generator.rb#L19-L30

**Comment by seport** [comment]
Thanks for your comments guys! I'll address these when I have a chance this weekend. 👍

**Comment by bbatsov** [comment]
@seport Just merged your other PR. Hopefully you'll have time to update this one as well.

**Comment by seport** [comment]
@bbatsov Sorry this took me so long! Hopefully this looks better. Let me know if any other changes need to be made!

**Comment by bbatsov** [comment]
It looks great! Thanks! 👍

**Pull request merged**